modules/navier_stokes/src/physics/WCNSFVTurbulencePhysics.C

```diff
@@ -0,0 +1,328 @@
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "WCNSFVTurbulencePhysics.h"
+#include "WCNSFVFlowPhysics.h"
+#include "WCNSFVFluidHeatTransferPhysics.h"
+#include "WCNSFVScalarTransportPhysics.h"
+#include "WCNSFVCoupledAdvectionPhysicsHelper.h"
+#include "NSFVAction.h"
+
+registerNavierStokesPhysicsBaseTasks("NavierStokesApp", WCNSFVTurbulencePhysics);
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_variable");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_fv_kernel");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_aux_variable");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_aux_kernel");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_material");
+
+InputParameters
+WCNSFVTurbulencePhysics::validParams()
+{
+  InputParameters params = NavierStokesPhysicsBase::validParams();
+  params += WCNSFVCoupledAdvectionPhysicsHelper::validParams();
+  params.addClassDescription("Define a turbulence model for a weakly-compressible Navier Stokes "
+                             "flow with a finite volume discretization");
+
+  MooseEnum turbulence_type("mixing-length none", "none");
+  params.addParam<MooseEnum>(
+      "turbulence_handling",
+      turbulence_type,
+      "The way turbulent diffusivities are determined in the turbulent regime.");
+  params += NSFVAction::commonTurbulenceParams();
+  params.transferParam<bool>(NSFVAction::validParams(), "mixing_length_two_term_bc_expansion");
+
+  // TODO Added to facilitate transition, remove default once NavierStokesFV action is removed
+  params.addParam<AuxVariableName>(
+      "mixing_length_name", "mixing_length", "Name of the mixing length auxiliary variable");
+
+  // Add the coupled physics
+  // TODO Remove the defaults once NavierStokesFV action is removed
+  // It is a little risky right now because the user could forget to pass the parameter and
+  // be missing the influence of turbulence on either of these physics. There is a check in the
+  // constructor to present this from happening
+  params.addParam<PhysicsName>(
+      "fluid_heat_transfer_physics",
+      "NavierStokesFV",
+      "WCNSFVFluidHeatTransferPhysics generating the heat advection equations");
+  params.addParam<PhysicsName>(
+      "scalar_transport_physics",
+      "NavierStokesFV",
+      "WCNSFVScalarTransportPhysics generating the scalar advection equations");
+
+  return params;
+}
+
+WCNSFVTurbulencePhysics::WCNSFVTurbulencePhysics(const InputParameters & parameters)
+  : NavierStokesPhysicsBase(parameters),
+    WCNSFVCoupledAdvectionPhysicsHelper(parameters, this),
+    _turbulence_model(getParam<MooseEnum>("turbulence_handling")),
+    _mixing_length_name(getParam<AuxVariableName>("mixing_length_name")),
+    _mixing_length_walls(getParam<std::vector<BoundaryName>>("mixing_length_walls"))
```

Comment:
This is hardcoded to mixing length. What about k-epsilon? Normally we should have a turbulent model selector as we do in most CFD codes

Reply:
We can rename when it gets into the syntax.

Reply:
renamed the attribute